pokemon-service/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/service/PokemonSpeciesServiceImpl.java

```diff
@@ -31,7 +31,7 @@
 
     @Override
     public void createPokemonSpecies(PokemonSpecies species) throws EvolutionChainTooLongException {
-        if (species.getEvolvesFrom() != null && evolutionChainNotLongerThan3(species.getEvolvesFrom(), species, true)) {
+        if (species.getEvolvesFrom() != null && !evolutionChainNotLongerThan3(species.getEvolvesFrom(), species, true)) {
```

Comment:
Why would we want the evolution chain not to be not longer than 3? If the implementation is broken, fix it there, don't change the semantics.

Reply:
Example: (preevolutions from the left) null < Pichu < Pikachu < Raichu
If I try to create Pikachu it throws an exception.
I think your implementation of method is good, and that you only forgot to negate the method result ("!")

Reply:
That would cover the broken implementation case, because the method contract says it returns true iff there is no evolution chain with the given species that would have more than 3 members.